src/sagas/account.js

```diff
@@ -37,13 +37,22 @@ export function * doSignup (action) {
 
   try {
     const response = yield minDelayCall(api.signup, formData);
-    response['nextRoute'] = formData.nextRoute;
 
-    // Persist token for future page loads
-    const tokenName = yield select(getTokenName);
-    new Cookies().set(tokenName, response.key, { path: '/' });
+    // If the response had an error status, but wasn't caught as an error
+    if (!!response.status && response.status > 299) {
+      yield put({
+        type: SIGNUP_ERROR, 
+        error: _.get(response, 'data'),
+      });
+    } else {
+      response['nextRoute'] = formData.nextRoute;
+
+      // Persist token for future page loads
+      const tokenName = yield select(getTokenName);
+      new Cookies().set(tokenName, response.key, { path: '/' });
 
-    yield put({ type: SIGNUP_SUCCESS, ...response });
+      yield put({ type: SIGNUP_SUCCESS, ...response });
+    }
   } catch (error) {
```

Comment:
Is the `try/catch` even needed when using `validateStatus`? Seems like the promise will never be rejected when `validateStatus: () => true`.

Reply:
@chris-allen  In the signup and login specific cases, what if there is an error with a missing nextRoute or if the Cookies fail in the set call?
For doPatchUser, I don't think it's needed.

Reply:
Sounds like a plain